### PR TITLE
es2015: bump eslint version

### PR DESCRIPTION
**Pull request by eszkadev**
- allow ES2015 in JS

### CODE REVIEW

File: browser/src/control/Control.JSDialogBuilder.js [hunk 1]
```diff
@@ -1824,7 +1824,7 @@ L.Control.JSDialogBuilder = L.Control.extend({
 		if (data.enabled === false || data.enabled === 'false')
 			$(listbox).attr('disabled', 'disabled');
 
-		$(listbox).change(function() {
+		$(listbox).change(() => {
 			if ($(this).val())
 				builder.callback('combobox', 'selected', data, $(this).val()+ ';' + $(this).children('option:selected').text(), builder);
 		});
```
Comment (Minion3665) [inline browser/src/control/Control.JSDialogBuilder.js:1830]: what is the `this` referring to here? I worry that by changing this to an arrow function you may have altered its meaning

**Review by Minion3665 — COMMENTED** [review]
From a quick look it seems fine except for the bit I commented on, which I'd need to test but looks to me like it could cause trouble...

File: browser/src/control/Control.JSDialogBuilder.js (hunk 1 above)
Comment (eszkadev) [reply browser/src/control/Control.JSDialogBuilder.js:1830]: ah, good catch. It is a jQuery callback where `this` was a `listbox` / target of the callback.
I think they do `.bind(listbox)` on the passed function object so it should work.
I checked it in Format -> Character -> Style. Seems to work.

Comment (eszkadev) [reply browser/src/control/Control.JSDialogBuilder.js]: good catch, fixed

Comment (Minion3665) [reply browser/src/control/Control.JSDialogBuilder.js]: can you bind arrow functions? I was under the impression that this isn't a thing you can do and from a quick test node seems to agree on this...
image

Comment (Minion3665) [reply browser/src/control/Control.JSDialogBuilder.js]: in contrast, here's what happens if you do the same thing without using an arrow function
image

**Review by Minion3665 — APPROVED** [review]

**Comment by vmiklos** [comment]
Nice! :-)

Do you plan to go further (I'm especially waiting for await in ES2007), or that'll be not for 24.04? Thanks.

**Comment by Minion3665** [comment]
> Nice! :-)
> 
> Do you plan to go further (I'm especially waiting for await in ES2007), or that'll be not for 24.04? Thanks.

@eszkadev I would also really like async/await... shall we do another?

**Comment by Minion3665** [comment]
image
https://caniuse.com/async-functions

caniuse looks fine

**Comment by vmiklos** [comment]
We discussed this on Thur and it should be OK to update to ES2017. However, when I try it with a trivial patch:

[await.txt](https://github.com/CollaboraOnline/online/files/14334256/await.txt)

then eslint fails like this:

> /home/vmiklos/git/collaboraonline/online/browser/src/map/Clipboard.js
>   719:21  error  Parsing error: Unexpected token blob
>
> ✖ 1 problem (1 error, 0 warnings)

@Minion3665 any idea what might be the problem? Thanks.

**Comment by hfiguiere** [comment]
`await` is only valid in an `async` block

**Comment by hfiguiere** [comment]
So you'd change

```
	_navigatorClipboardGetTypeCallback: function(clipboardContent, blob, type) {
```
to
```
	_navigatorClipboardGetTypeCallback: async function(clipboardContent, blob, type) {
```

**Comment by vmiklos** [comment]
Thanks, see https://github.com/CollaboraOnline/online/pull/8319

**Pull request merged**